Refactor/finish coroutines for mongoDB

Al-Qassim opened the pull request.

Al-Qassim committed:
make simple coroutine with runBlocking

Al-Qassim committed:
just save

Al-Qassim committed:
fully working coroutine

Al-Qassim committed:
improve naming and loading message

Al-Qassim committed:
- rename tryCall to makeRequest in BaseView.
- use onSuccess function instead of returning a boolean.

Al-Qassim committed:
- rename exceptionToBeCaught to caughtException

ZainabAlnajjar commented on build.gradle.kts:
delete the unused dependencies

ZainabAlnajjar commented on src/main/kotlin/ui/projectDetailsView/EditProjectView.kt:
```diff
@@ -20,37 +20,46 @@ class EditProjectView(
 
-    private fun fetchProject(projectId: UUID) {
+    private fun printOptions() {
+        cliPrinter.printHeader("Edit Project: ${currentProject.title}")
```
the header is not an option 🤔

Al-Qassim committed:
made the loading message appear faster

Al-Qassim committed:
improve naming and moved printing the header outside the print options function

Al-Qassim committed:
print the line break after the loop ends

Al-Qassim committed:
- put onLoading message for all views
- optimized the logsView a little bit

poula22 commented on src/main/kotlin/data/dataSources/mongoDBDataSource/DataBaseConnection.kt:
```diff
@@ -15,7 +15,7 @@ object DatabaseConnection {
     private val mongoUri: String = System.getenv("MONGO_URI")?: throw IllegalStateException("MONGO_URI environment variable is missing")
 
-    private val client = MongoClients.create(mongoUri)
+    private val client = MongoClient.create(mongoUri)
```
by lazy{ MongoClient.create(mongoUri) }

poula22 commented on src/main/kotlin/main.kt:
```diff
@@ -18,4 +22,8 @@ fun main() {
     ui.start()
+
+    // todo: make the app shutdown in a better way
+    DatabaseConnection.close()
```
??

Al-Qassim committed:
- solve a bug in the table
- no need to check for admin ManageMatesView
- check admin in the CreateMateUseCase

ElhamFadel commented on src/main/kotlin/ui/BaseView.kt:
```diff
@@ -1,24 +1,74 @@
 import ui.cliPrintersAndReaders.CLIPrinter
 
-open class BaseView(
+abstract class BaseView(
```
```suggestion
abstract class BaseRequestHandler(
```
Maybe this name not good enough

I think using name that address what this class do like `BaseRequestHandler`

Al-Qassim replied:
`RequestHandler` is good for me

Al-Qassim committed:
- solving a bug in the WelcomeView
- solving a bug in the csvStorageModule
- creating only one coroutine in the base view

Al-Qassim committed:
renamed blockWithLoadingLoop to startLoadingAnimation

Al-Qassim committed:
rename BaseView to RequestHandler
